### DISCUSSION TITLE
Problem Adding Access Policy for KeyVault - two principals?

### DISCUSSION THREAD

**Discussion by dazinator** [polymind-inc/acmebot#437] [Q&A] [answered]
I deployed via the ARM template.
I'm just on the last step of adding an access policy to keyvault for certificate management.
When searching for the principal, there are two possible candidates:

![image](https://user-images.githubusercontent.com/3176632/148559446-6dd22d99-ae39-4f01-8afc-f80f55f14637.png)

Is this a bug? How do I know which one to use?

It's a bit odd to me because I can only see one function app in the resource group, yet on the ARM deployment I can see two "sites" were created:

![image](https://user-images.githubusercontent.com/3176632/148559940-02d747e2-bcb3-4b0f-a252-6247f5b599e1.png)

**Comment by dazinator** [comment]
In terms of working out which prinicpal is the correct one, I think the following works:

1. Go into the function app, click oon "Identity" tab and get the "Object Id"
  
   ![image](https://user-images.githubusercontent.com/3176632/148560941-6accbb6d-8ae2-4077-a69e-262707bedbaf.png)

2. When creting the access policy in key vault, on the prinipal search UI, search for the object id.
   
   ![image](https://user-images.githubusercontent.com/3176632/148561090-1f2d6c51-1672-43ac-b425-ed469f85090e.png)

It seems to return only of the principals so I am assuming this is correct one.
As to why there are two, I am not sure.

**Comment by shibayan** [comment, marked as answer]
I just tried deploying from ARM Template, but I couldn't reproduce it. The principal creation is done inside Azure, so it is most likely a bug on the Azure side.

The way to check the correct Principal is as you wrote.

**Reply by dazinator** [reply]
Ok, thanks.

**Comment by rvdwegen** [comment]
I ran into the same thing and in my case one of them was an app registration in AzureAD
